model/labels/regexp.go

```diff
@@ -63,13 +63,20 @@ func NewFastRegexMatcher(v string) (*FastRegexMatcher, error) {
 		// available, even if the string matcher is faster.
 		m.matchString = m.stringMatcher.Matches
 	} else {
-		parsed, err := syntax.Parse(v, syntax.Perl)
+		var parsed *syntax.Regexp
+		var err error
+		if v == ".*" { // special case for .* to match everything
+			flags := syntax.ClassNL | syntax.OneLine | syntax.PerlX | syntax.UnicodeGroups | syntax.DotNL
+			parsed, err = syntax.Parse(v, flags)
```

Comment:
My expectation is we use the same flags for all regexps.

I asked this before at https://github.com/prometheus/prometheus/pull/14505/files#r1709714579, but I don't think it was answered.